### ISSUE TITLE
opencart_1  | Error executing 'postInstallation': Cannot read property '1' of null

### ISSUE THREAD

**Issue by fsc7** [bitnami/bitnami-docker-opencart#47]
After stopping the container and starting up again I get the following error:

opencart_1  | nami    INFO  Initializing apache
opencart_1  | apache  INFO  ==> Reconfiguring PID file location...
opencart_1  | nami    INFO  apache successfully initialized
opencart_1  | nami    INFO  Initializing php
opencart_1  | nami    INFO  php successfully initialized
opencart_1  | nami    INFO  Initializing opencart
opencart_1  | Error executing 'postInstallation': Cannot read property '1' of null
ecommerce_opencart_1 exited with code 1

**Comment by javsalgar** [comment]
Hi,

Thanks for reporting the issue. I was able to reproduce the issue. The problem is when restoring the persisted configuration in `config.php`. It will try extract the IP address from the file but it is not considering the case where domain names like `localhost`, and that's why it fails.

We will work on fixing the issue. As a workaround you can manually edit the config.php file:

```
define('HTTP_SERVER', 'http://localhost/');
```

Change it to IP address.

```
define('HTTP_SERVER', 'http://127.0.0.1/');
```

**Comment by fsc7** [comment]
Thanks. It solved the problem.

**Comment by danielbjornadal** [comment]
Any update on this ? 
The workaround works, but the error is still there.

**Comment by juan131** [comment]
Hi @danielbjornadal 

We have at task included in our backlog. As the impact of the issue is limited, there are no immediate plans to work on it. Nevertheless, it will be considered for future application versions.

We will update this thread as soon as we have more news.

**Comment by atomicnation** [comment]
I'm facing the same issue when loading the bitnami/opencart image from docker. Any ideas on how could I solve it?

**Comment by marcosbc** [comment]
Hi, we have just released [OpenCart `3.0.3-6-debian-10-r65`](https://github.com/bitnami/bitnami-docker-opencart#303-6-debian-10-r65) which includes major changes from previous releases. Now, source code for the initialization logic is published in GitHub in the [`rootfs` folder](https://github.com/bitnami/bitnami-docker-opencart/tree/master/3/debian-10/rootfs) and also introduces significant changes to the initialization approach.

In particular, this issue should not appear anymore due to the improved logic. Please let us know if we can consider this solved!
